### PR TITLE
Replace gon macos signing with notarytool

### PR DESCRIPTION
**Pull request by ddelnano**
Summary: Replace gon macos signing with notarytool

Relevant Issues: #1993

Type of change: /kind bugfix

Test Plan: Read [a](https://github.com/mitchellh/gon/issues/40#issuecomment-1261128054) few [examples](https://successfulsoftware.net/2023/04/28/moving-from-altool-to-notarytool-for-mac-notarization/) of migrated cli invocations, however, seeing a successful cli release is necessary to fully test this

### CODE REVIEW

File: ci/cli_merge_sign.sh [hunk 1]
```diff
@@ -48,7 +48,13 @@ done
 # Create a universal binary.
 lipo -create -output cli_darwin_universal cli_darwin_arm64 cli_darwin_amd64
 
-gon ci/gon.hcl
+export TEAM_ID="SZCNTABEXY"
+export APP_IDENTITY="Developer ID Application: Pixie Labs Inc. (${TEAM_ID})"
+codesign -f -v --timestamp --options runtime -s "${APP_IDENTITY}" cli_darwin_amd64 cli_darwin_arm64 cli_darwin_universal
```
Comment (ddelnano) [inline ci/cli_merge_sign.sh:53]: This was modeled off how gon runs codesign ([source](https://github.com/mitchellh/gon/blob/2d4f161ccecd1aae878f4416d5ccd622b1b01fdb/sign/sign.go#L66-L73))

**Comment by vihangm** [comment]
I'm wondering if we should switch to https://github.com/Bearer/gon which seems to be a maintained fork of the original and switches over to `notarytool`

**Comment by ddelnano** [comment]
Using that fork is fine with me. I'll update this PR to take that approach. Unfortunately the developer account we use has expired and is blocking finishing off the testing of this.

**Comment by ddelnano** [comment]
Closing since we will be proceeding with #1997.

**Pull request closed without merging**